quantum/encoder.c

```diff
@@ -138,25 +145,29 @@ void last_encoder_activity_trigger(void);
 void encoder_state_raw(uint8_t* slave_state) { memcpy(slave_state, &encoder_value[thisHand], sizeof(uint8_t) * NUMBER_OF_ENCODERS); }
 
 void encoder_update_raw(uint8_t* slave_state) {
-    bool changed = false;
-    for (uint8_t i = 0; i < NUMBER_OF_ENCODERS; i++) {
-        uint8_t index = i + thatHand;
-        int8_t  delta = slave_state[i] - encoder_value[index];
-        while (delta > 0) {
-            delta--;
-            encoder_value[index]++;
-            changed = true;
-            encoder_update_kb(index, ENCODER_COUNTER_CLOCKWISE);
-        }
-        while (delta < 0) {
-            delta++;
-            encoder_value[index]--;
-            changed = true;
-            encoder_update_kb(index, ENCODER_CLOCKWISE);
+    uint8_t changed = 0;
+    uint8_t index   = thatHand;
+    for (uint8_t i = 0; i < NUMBER_OF_ENCODERS; i++, index++) {
+#ifndef ENCODER_DIRECTION_FLIP
+        int8_t delta = slave_state[i] - encoder_value[index];
+#else
+        int8_t delta = encoder_value[index] - slave_state[i];
+#endif
+        int8_t incr = (delta >> 7); /* if delta <= -1 then incr = -1 else incr = 1 */
+        bool   cw = incr & 1;       /* if incr < 0 then cw = false else cw = true */
+        incr |= 1;
+        changed |= incr;
+        while (delta != 0) {
+            delta -= incr;
+            encoder_update_kb(index, cw);
         }
+        encoder_value[index] = slave_state[i];
     }
 
+    // (delta < 0) -> ENCODER_CLOCKWISE = true/1
+    // (delta > 0) -> ENCODER_COUNTER_CLOCKWISE = false/0
+
     // Update the last encoder input time -- handled external to encoder_read() when we're running a split
-    if (changed) last_encoder_activity_trigger();
+    if (changed != 0) last_encoder_activity_trigger();
```

Comment:
This should still work, without the change. 
```suggestion
    if (changed) last_encoder_activity_trigger();
```

Reply:
Depending on the type, this still may result in slightly smaller code, though.
Will need confirmation from @jurgen-kluft.

Reply:
Tested locally, compiles with no difference in size. So mostly just a preference thing, I guess.